web-api/send-migration-segment-messages.ts

```diff
@@ -0,0 +1,68 @@
+import { DescribeTableCommand, DynamoDBClient } from '@aws-sdk/client-dynamodb';
+import { addToQueue } from '../shared/admin-tools/aws/sqsHelper';
+import { shuffle } from 'lodash';
+
+// We found that 200 seems to be a good number to choose.  Anything too high and migrations stop working well.
+const SEGMENT_SIZE = 200;
+
+const [ENV] = process.argv.slice(2);
+
+if (!ENV) {
+  throw new Error('Please provide an environment.');
+}
+
+if (!process.env.AWS_ACCOUNT_ID) {
+  throw new Error('Please set AWS_ACCOUNT_ID in your environment.');
+}
+
+if (!process.env.SOURCE_TABLE) {
+  throw new Error('Please set SOURCE_TABLE in your environment.');
+}
+
+const QueueUrl = `https://sqs.us-east-1.amazonaws.com/${process.env.AWS_ACCOUNT_ID}/migration_segments_queue_${ENV}`;
+
+const dynamodb = new DynamoDBClient({
+  maxAttempts: 10,
+  region: 'us-east-1',
+});
+
+const getItemCount = async (): Promise<number> => {
+  let count = 0;
+  const describeTableCommand = new DescribeTableCommand({
+    TableName: process.env.SOURCE_TABLE,
+  });
+  try {
+    const { Table } = await dynamodb.send(describeTableCommand);
+    if (
+      Table &&
+      'ItemCount' in Table &&
+      Table.ItemCount &&
+      Table.ItemCount > 0
+    ) {
+      count = Table.ItemCount;
+    }
+  } catch (e) {
+    console.error('Error retrieving dynamo item count.', e);
+  }
+  return count;
+};
+
+// eslint-disable-next-line @typescript-eslint/no-floating-promises
+(async () => {
+  const itemCount = await getItemCount();
+
+  const totalSegments = Math.max(1, Math.ceil(itemCount / SEGMENT_SIZE));
+
+  const segments = shuffle(
+    new Array(totalSegments).fill(null).map((_v, i) => ({
+      segment: i,
+      totalSegments,
+    })),
+  );
+
+  const { failed } = await addToQueue({ QueueUrl, messages: segments });
+  if (failed.length) {
+    // TODO: bail?
+    // process.exit(1);
```

Comment:
Should this exit here?

Reply:
Thanks for mentioning this.

We weren't previously detecting failures, but we as you can see, we are doing so now. It doesn't look like we can easily retry the failed segment messages, though; the `BatchResultErrorEntry` errors do not include the original message contents.

I think we should exit. If we don't exit here, we could potentially switch over to a destination table with incomplete data and then delete the source table in the cleanup step. That would be real bad.